contribs/ev/src/main/java/org/matsim/contrib/ev/discharging/DriveDischargingHandler.java

```diff
@@ -104,6 +111,7 @@ public void handleEvent(VehicleLeavesTrafficEvent event) {
 	// (for instance, AUX discharging and battery charging modifies charge outside event handling
 	// (as MobsimAfterSimStepListeners)
 	//TODO In the long term, it will be safer to move the discharging procedure to a MobsimAfterSimStepListener
+	// -> has been implemented now as a hack /sebhoerl
```

Comment:
While this change will definitely help with the order of events (which is good!), I was thinking about updating the SOC in the main thread (outside the event handling threads) to avoid race conditions (and also to have events deterministically ordered).

I will create a PR for moving the whole logic outside the event handling.